mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs

```diff
@@ -775,7 +778,9 @@ private Task<TResult> GetTaskForResult(TResult result)
                     (typeof(TResult) == typeof(Byte) && default(Byte) == (Byte)(object)result) ||
                     (typeof(TResult) == typeof(SByte) && default(SByte) == (SByte)(object)result) ||
                     (typeof(TResult) == typeof(Char) && default(Char) == (Char)(object)result) ||
+#if !MONO
                     (typeof(TResult) == typeof(Decimal) && default(Decimal) == (Decimal)(object)result) ||
+#endif
                     (typeof(TResult) == typeof(Int64) && default(Int64) == (Int64)(object)result) ||
```

Comment:
Not sure about this change, just copied it from https://github.com/mono/mono/commit/b3881a418f1391a33c1b709f1113574b74ae85ae#diff-80de53f8dc1f0a25e4dec310dfe1a3c7R784